ratis-examples/src/main/java/org/apache/ratis/examples/filestore/FileStoreStateMachine.java

```diff
@@ -40,14 +41,18 @@
 import org.apache.ratis.thirdparty.com.google.protobuf.ByteString;
 import org.apache.ratis.thirdparty.com.google.protobuf.InvalidProtocolBufferException;
 import org.apache.ratis.util.FileUtils;
+import org.apache.ratis.util.JavaUtils;
 
 import java.io.File;
 import java.io.IOException;
 import java.util.Objects;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
 
 public class FileStoreStateMachine extends BaseStateMachine {
   private final SimpleStateMachineStorage storage = new SimpleStateMachineStorage();
+  private final ConcurrentMap<LogEntryProto, DataStream> dataStreamToLogEntryMap = new ConcurrentHashMap<>();
```

Comment:
Seems we only put entry in dataStreamToLogEntryMap, but never used.

Reply:
See discussion in https://github.com/apache/incubator-ratis/pull/298#discussion_r533447995.

Basically I am not sure whether my current implementation is enough. If so I can remove the link() and this data structure.

Reply:
Yes, I see the discussion.  Because we have the path in the header, dataStreamToLogEntryMap  is never used, so I think we do not need it here, you can add a comment in the link method. @szetszwo What do you think?

Reply:
Let's remove this map since it is currently not useful.